junit-platform-surefire-provider/src/main/java/org/junit/platform/surefire/provider/RunListenerAdapter.java

```diff
@@ -83,36 +82,19 @@ else if (testIdentifier.isTest()) {
 	}
 
 	private SimpleReportEntry createReportEntry(TestIdentifier testIdentifier, Optional<Throwable> throwable) {
-		Optional<String> className = getClassName(testIdentifier);
+		final TestDescriptor.LegacyReportingInfo legacyReportingInfo = testIdentifier.getLegacyReportingInfo();
+		Optional<String> className = legacyReportingInfo.getClassName();
+		final String methodName = legacyReportingInfo.getMethodName().orElse("");
```

Comment:
If you do the rename I requested above, this local variable should then be called `name`.